### ISSUE TITLE
Fix readystate during seeking

### ISSUE THREAD

**Issue by varumugam123** [Metrological/WebKitForWayland#821]
While seeking out-of-bound, we receive "playing" event... On chrome & firefox, this is not happening. Attaching a simple case of this here
[play_pause_out_of_range_test.html.txt](https://github.com/WebPlatformForEmbedded/WPEWebKit/files/8371420/play_pause_out_of_range_test.html.txt)

Below force updation of readyState to HAVE_METADATA handles this case.
```
diff --git a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
index 1462ec4cbc3f..5b7d0bf696ca 100644
--- a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
@@ -187,6 +193,7 @@ void MediaPlayerPrivateGStreamerMSE::seek(const MediaTime& time)
             toString(current).utf8().data(), toString(time).utf8().data(),
             url().string().utf8().data());

+    MediaPlayer::ReadyState oldReadyState = m_readyState;
     MediaTime previousSeekTime = m_seekTime;
     m_seekTime = time;

@@ -197,6 +204,11 @@ void MediaPlayerPrivateGStreamerMSE::seek(const MediaTime& time)
     }

     m_isEndReached = false;
+    if ((m_isSeeking && oldReadyState > MediaPlayer::ReadyState::HaveMetadata) {
+        m_readyState = MediaPlayer::ReadyState::HaveMetadata;
+        m_player->readyStateChanged();
+    }
+
     GST_DEBUG_OBJECT(pipeline(), "m_isSeeking=%s, m_seekTime=%s", boolForPrinting(m_isSeeking), toString(m_seekTime).utf8().data());
 }
```

Note: this has dependency on https://github.com/WebPlatformForEmbedded/WPEWebKit/issues/816, without #816 the playing events are still sent unconditionally from HTMLMediaElement::setPlaying()

**Comment by eocanha** [comment]
I see the problem that the test tries to reproduce. The test buffers [0, 11.99] and, without having started playing, seeks to 12.99 (unbuffered). Right after that, sets play() and pause() very quickly in the same main loop iteration (so the effective state should be paused). The "playing" event is emitted after seek, and it shouldn't by two reasons: the player state should be paused, and there's no data to start playback.

The proposed patch appears to fix the test case (by avoiding the "playing" event, since the state is now clearly HaveMetadata and that's not enough for playback). However, it breaks another use case that was working before:

Set the player to play() when still at 0 and with the [0, 11.99] data loaded ("playing" is emitted, but that's irrelevant for this explanation), wait a while (200ms), then seek to a buffered range (eg: to 1). Since the target seek position is buffered and the player is playing, nothing should stop it after seek and no "playing" event should be emitted when the seek completes. However, because of the readyState resetting to HaveMetadata done by your patch, the player now thinks that it's coming out from a stall situation and emits the "playing" event. It shouldn't since the seek was to a buffered range (no stall).

**Comment by eocanha** [comment]
I worked on commit https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/a00d108bcd2ddb6886411715c2dc897aef60ce72 (branch [eocanha-debug-115](https://github.com/WebPlatformForEmbedded/WPEWebKit/tree/eocanha/eocanha-debug-115)) that works for both cases. I still need to see if it's upstreamable.

**Comment by eocanha** [comment]
It's not upstreamable. I've submitted the change as PR https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/825 and merged it on wpe-2.28.
